collector/test/ConnTrackerTest.cpp

```diff
@@ -1651,80 +1651,143 @@ TEST(ConnTrackerTest, TestConnectionStats) {
-  tracker.CloseNormalizedConnections(&old_state, &delta);
-
-  EXPECT_THAT(old_state, IsEmpty());
-  EXPECT_THAT(delta, expected_delta);
-}
+TEST(ConnTrackerTest, TestExternalIPsConfigChangeEnableEgress) {
+  Endpoint ingress_local(IPNet(Address()), 80);
+  Endpoint ingress_remote(Address(223, 42, 0, 1), 0);
+  Endpoint ingress_remote_normalized(Address(255, 255, 255, 255), 0);
 
-TEST(CloseNormalizedConnectionsTest, UnnormalizedConnectionsAreKept) {
-  ConnectionTracker tracker;
+  Endpoint egress_local = Endpoint();
+  Endpoint egress_remote(Address(223, 42, 0, 2), 443);
+  Endpoint egress_remote_normalized(Address(255, 255, 255, 255), 443);
 
-  Endpoint a(Address(192, 168, 0, 1), 80);
-  Endpoint b(Address(192, 168, 1, 10), 9999);
+  Connection conn_ingress("xyz", ingress_local, ingress_remote, L4Proto::TCP, true);
+  Connection conn_ingress_normalized("xyz", ingress_local, ingress_remote_normalized, L4Proto::TCP, true);
+  Connection conn_egress("xyz", egress_local, egress_remote, L4Proto::TCP, false);
+  Connection conn_egress_normalized("xyz", egress_local, egress_remote_normalized, L4Proto::TCP, false);
 
-  Connection conn("xyz", a, b, L4Proto::TCP, true);
   int64_t connection_time = 990;
 
-  ConnMap old_state = {{conn, ConnStatus(connection_time, true)}};
-  ConnMap delta;
-  ConnMap expected_old_state = {{conn, ConnStatus(connection_time, true)}};
-
-  tracker.CloseNormalizedConnections(&old_state, &delta);
-
-  EXPECT_THAT(old_state, expected_old_state);
-  EXPECT_THAT(delta, IsEmpty());
-}
-
-TEST(ConnTrackerTest, TestCloseExternalUnnormalizedConnections) {
-  ConnectionTracker tracker;
-
-  Endpoint a(Address(192, 168, 0, 1), 80);
-  Endpoint b(Address(11, 168, 1, 10), 9999);
-
-  Connection conn("xyz", a, b, L4Proto::TCP, true);
-  int64_t connection_time = 990;
-
-  ConnMap old_state = {{conn, ConnStatus(connection_time, true)}};
-  ConnMap delta;
-  ConnMap expected_delta = {{conn, ConnStatus(connection_time, false)}};
-
-  tracker.CloseExternalUnnormalizedConnections(&old_state, &delta);
-
-  EXPECT_THAT(old_state, IsEmpty());
-  EXPECT_THAT(delta, expected_delta);
-}
-
-TEST(CloseExternalUnnormalizedConnectionsTest, InternalConnectionsAreKept) {
-  ConnectionTracker tracker;
-
-  Endpoint a(Address(192, 168, 0, 1), 80);
-  Endpoint b(Address(192, 168, 1, 10), 9999);
-
-  Connection conn("xyz", a, b, L4Proto::TCP, true);
-  int64_t connection_time = 990;
-
-  ConnMap old_state = {{conn, ConnStatus(connection_time, true)}};
-  ConnMap delta;
-  ConnMap expected_old_state = {{conn, ConnStatus(connection_time, true)}};
-
-  tracker.CloseExternalUnnormalizedConnections(&old_state, &delta);
-
-  EXPECT_THAT(old_state, expected_old_state);
-  EXPECT_THAT(delta, IsEmpty());
+  struct TestStep {
+    ExternalIPsConfig::Direction previous_direction;
+    ExternalIPsConfig::Direction new_direction;
+    ConnMap old_state;
+    ConnMap resulting_old_state;
+    ConnMap expected_delta;
+  } testSteps[] = {
+      {// No change (enabled)
```

Comment:
There are 16 possible combinations for `previous_direction` and `new_direction`. It would not be unreasonable to have all of them. I added a couple of them in the PR I opened.

Reply:
I cherry-picked/adapted the tests you added on your PR. I will prioritize some manual testing over adding all the combinations for now.

Reply:
Okay.